From 30455efd7dc4d461d38f65849b3cd482787ad90d Mon Sep 17 00:00:00 2001
From: Kyle Brenneman <kbrenneman@nvidia.com>
Date: Thu, 1 May 2025 12:30:03 -0600
Subject: [PATCH 2/3] Don't close the syncfd in WaitImplicitFence.

In WaitImplicitFence, we pass the syncfd to eglCreateSync, which takes
ownership of it, so we shouldn't close it ourselves.
---
 src/x11/x11-window.c | 1 -
 1 file changed, 1 deletion(-)

diff --git a/src/x11/x11-window.c b/src/x11/x11-window.c
index ae7bc17..48de96c 100644
--- a/src/x11/x11-window.c
+++ b/src/x11/x11-window.c
@@ -1710,7 +1710,6 @@ static EGLBoolean WaitImplicitFence(EplDisplay *pdpy, X11ColorBuffer *buffer)
     if (fd >= 0)
     {
         success = WaitForSyncFDGPU(pdpy->priv->inst, fd);
-        close(fd);
     }
 
     if (success)
-- 
2.43.0

